src/lib.rs

```diff
@@ -1,38 +1,286 @@
+```
+use bevy::prelude::*;
+use bevy_enhanced_input::prelude::*;
 
-[`Conditions`](input_condition) define how an action activates. We also provide built-in conditions, such as [`Press`],
-[`Release`], [`Hold`], etc. You can also add your own by implementing [`InputCondition`].
+let mut app = App::new();
+app.add_plugins((MinimalPlugins, EnhancedInputPlugin));
+```
 
-# Quick start
+## Defining actions
 
-We provide a [`prelude`] module, which exports most of the typically used traits and types.
+Actions represented by unit structs that implement the [`InputAction`] trait, which also requires
+[`Debug`] which we use for logging. You can use the provided derive macro for this.
+
+```
+# use bevy::prelude::*;
+# use bevy_enhanced_input::prelude::*;
+#[derive(Debug, InputAction)]
+#[input_action(output = bool)]
+struct Jump;
+
+#[derive(Debug, InputAction)]
+#[input_action(output = Vec2)]
+struct Move;
+```
+
+The `output` defines the [`InputAction::Output`] type and is the only required parameter. See the [`InputAction`]
+documentation for all parameters.
+
+## Defining input contexts
+
+Actions are bound to input contexts. Depending on your type of game, you may have a single global context or multiple
+contexts for different gameplay states. Consider contexts as sets of gameplay actions.
+
+Input contexts are represented by unit structs that implement the [`InputContext`] trait. You can use the provided derive
+macro for this. Unlike actions, all contexts need to be registered in the app using [`InputContextAppExt::add_input_context`].
+
+```
+# use bevy::prelude::*;
+# use bevy_enhanced_input::prelude::*;
+# let mut app = App::new();
+# app.add_plugins(EnhancedInputPlugin);
+app.add_input_context::<OnFoot>();
+
+#[derive(InputContext)]
+struct OnFoot;
+```
+
+No required parameters needed. See the [`InputContext`] documentation for all parameters.
+
+## Binding actions
+
+While input contexts are defined statically at compile time, bindings must be assigned at runtime. They're stored inside the
+[`Actions<C>`] component, where `C` is an input context. Contexts becomes active only when its component exists on an
+entity. You can attach multiple [`Actions<C>`] components to a single entity.
+
+To bind actions, use [`Actions::bind`] followed by one or more [`ActionBinding::to`] calls to define inputs. You can pass any
+input type that implements [`IntoBindings`], including tuples for multiple inputs (similar to [`App::add_systems`] in Bevy).
+All assigned inputs will be treated as "any of". See [`ActionBinding::to`] for details.
+
+```
+# use bevy::prelude::*;
+# use bevy_enhanced_input::prelude::*;
+let mut actions = Actions::<OnFoot>::default();
+// The action will trigger when space or gamepad south button is pressed.
+actions.bind::<Jump>().to((KeyCode::Space, GamepadButton::South));
+# #[derive(InputContext)]
+# struct OnFoot;
+# #[derive(Debug, InputAction)]
+# #[input_action(output = bool)]
+# struct Jump;
+```
+
+### Input modifiers
```

Comment:
Framing and flow of this section is quite odd. I'm not immediately sure what to suggest.